src/api/app/controllers/webui/staging/projects_controller.rb

```diff
@@ -25,6 +25,10 @@ def create
       def show
         @staging_project = @staging_workflow.staging_projects.find_by(name: params[:project_name])
         @project = @staging_workflow.project
+        @groups_hash = {}
+        Group.find_each do |group|
```

Comment:
What about writing that into the cache in a function in group?

```rb
def group_cache
  Rails.cache.fetch(Group.all.cache_key) do
    result = {}
    Group.find_each do |group|
      result[group.title] = group
    end
    result
  end
end
```

Reply:
We should move this code to a place where it can be shared. It is now repeated. :sob: